Add continuous question

nikosbosse opened the pull request:
This PR adds continuous questions. When reviewing, please bear in mind that I'm a terrible python programmer, so very happy about any edits/rewrites etc. you might make. 
Have to double check the math when converting scaled distributions (all metaculus distributions are mapped to [0,1] and have to be transformed back...)

edomt commented:
Great improvement @nikosbosse! 

I've fixed a few things. The one remaining problem is that the implementation of the trigger for continuous questions (lines 267 to 275) seems to allow for quite small changes. For example right now the following alert would be tweeted:

`{'text': 'How much foreign aid will the US provide Ukraine in 2022?\n\nCommunity prediction: 5119 million\n⬆️ +150 million in the last 48 hours\nhttps://www.metaculus.com/questions/10024/us-foreign-aid-to-ukraine/', 'chart': '/tmp/tmpdacsvj0r.png'}`

which is only a change of about 3%. I'm not sure if you know why?

nikosbosse commented on get_predictions.py:
```diff
@@ -132,11 +132,18 @@ def add_tweet(
+                    change_formatted = f"{added_sign}{round(change, 2)}"
+                    current_pred_formatted = str(round(current_prediction, 2))
+                elif current_prediction >= 1e7:
+                    change_formatted = f"{added_sign}{int(change / 1e7)} million"
```
shouldn't this be 1e6?

edomt replied:
…yes, it should 😞

nikosbosse commented:
Thanks a lot, @edomt! And good catch with the date questions - just didn't think about them.
Regarding the small updates: we should probably change the factor from 2 to something like 5 or 10 (and maybe assign it to a variable instead of just hard-coding it in there). If threshold is 0.1 and factor is 5, then an update would trigger if the median changes by at least 50% of the distance from old median to old 25% or old 75% percentile. 
I also need to update the way we map from the [0, 1] interval to the actual values

nikosbosse commented:
I updated the threshold and the calculation (should be correct now) and also updated the logic when to tweet here: https://github.com/nikosbosse/Metaculus-Twitter-bot/pull/22

Currently on a different computer, so can't test it and made edits through the github UI (always a good sign everything is definitely full of errors...). Can test later, but also happy for you to merge if you prefer.

edomt commented:
Looks good to me - no error on a dry run, and the only raised alert is `[{'text': 'Will WHO declare the spread of monkeypox a Public Health Emergency of International Concern before 2023?\n\nCommunity prediction: 30%\n⬇️ -30% in the last 24 hours\nhttps://www.metaculus.com/questions/10977/who-declares-that-monkeypox-is-a-pheic/', 'chart': '/tmp/tmplu9cfwft.png'}]`
